core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts

```diff
@@ -139,7 +140,11 @@ export class DotExperimentsConfigurationGoalSelectComponent implements OnInit, O
             primary: new FormGroup({
                 name: new FormControl('', {
                     nonNullable: true,
-                    validators: [Validators.required, Validators.maxLength(this.maxNameLength)]
+                    validators: [
+                        Validators.required,
+                        Validators.maxLength(this.maxNameLength),
+                        DotValidators.noWhitespace
```

Comment:
```suggestion
                        Validators.pattern(^[^\s]*$)
```

Instead of create a full validator, you can use regex to validate no white spaces.

Reply:
I moved the regex validation to the custom validator; the thing is that we need a custom message for that so we use `DOT_ERROR_MESSAGES.whiteSpaceOnlyMgs` if we use the `pattern` we will use the `pattern` key in the message map for this angular validator.